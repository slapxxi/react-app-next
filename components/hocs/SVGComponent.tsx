import { ComponentProps, ComponentType } from 'react';

interface SVGProps extends ComponentProps<'g'> {
  size: number;
  className?: string;
}

type Config = React.SVGAttributes<SVGElement>;

function SVGComponent<Props extends SVGProps>(
  Component: React.ComponentType<Props>,
  config?: Config,
): React.ComponentType<Props> {
  let SVGComp: ComponentType<Props> = (props: Props) => {
    let { size, className, ...rest } = props;

    return (
      <svg width={size} height={size} className={className} {...config}>
        <Component {...rest} className={className} />
      </svg>
    );
  };

  SVGComp.displayName = 'SVGComponent';

  return SVGComp;
}

export default SVGComponent;
