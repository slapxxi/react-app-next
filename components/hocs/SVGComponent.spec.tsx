/** @jsx jsx */
import { css, jsx } from '@emotion/core';
import { render } from 'react-testing-library';
import SVGComponent from './SVGComponent';

it('renders', () => {
  let Component = SVGComponent(() => <circle cx="10" cy="10" r="5" />);
  let { container } = render(<Component size={24} />);

  expect(container.firstChild).toMatchSnapshot();
});

it('accepts configuration', () => {
  let Component = SVGComponent(() => <circle cx="10" cy="10" r="5" />, {
    viewBox: '0 0 10 10',
  });
  let { container } = render(<Component size={24} />);

  expect(container.firstChild).toMatchSnapshot();
});

it('accepts css property', () => {
  let Component = SVGComponent(() => <circle cx="10" cy="10" r="5" />);
  let { container } = render(
    <Component
      size={24}
      css={css`
        fill: red;
      `}
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});
