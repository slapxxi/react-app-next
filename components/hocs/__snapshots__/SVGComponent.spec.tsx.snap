// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts configuration 1`] = `
<svg
  height="24"
  viewBox="0 0 10 10"
  width="24"
>
  <circle
    cx="10"
    cy="10"
    r="5"
  />
</svg>
`;

exports[`accepts css property 1`] = `
<svg
  class="css-18259do"
  height="24"
  width="24"
>
  <circle
    cx="10"
    cy="10"
    r="5"
  />
</svg>
`;

exports[`renders 1`] = `
<svg
  height="24"
  width="24"
>
  <circle
    cx="10"
    cy="10"
    r="5"
  />
</svg>
`;
